controllers/package_controller.go

```diff
@@ -80,9 +80,11 @@ func RegisterPackageReconciler(mgr ctrl.Manager) (err error) {
 	if err != nil {
 		return fmt.Errorf("creating discovery client: %s", err)
 	}
+	kvc := bundle.NewKubeVersionClient(discovery)
 	puller := artifacts.NewRegistryPuller()
+	registryClient := bundle.NewRegistryClient(log, puller)
 	bundleClient := bundle.NewPackageBundleClient(mgr.GetClient())
-	bundleManager := bundle.NewBundleManager(log, discovery, puller, bundleClient)
+	bundleManager := bundle.NewBundleManager(log, kvc, registryClient, bundleClient)
```

Comment:
Should we create a factory for this repo where all new creations happen in? The idea would be to do the instantiation of new objects in one place. We could follow a similar pattern from eks-anywhere. Similar to this: https://github.com/aws/eks-anywhere/blob/main/pkg/dependencies/factory.go. I found it very helpful for managing object creations.

Reply:
IMO, the CLI took dependency injection a bit too far and that is more of the symptom of the problem rather than a solution. Factories are supposed to hide the implementation and that really doesn't do that. I think creating that factory is a good step for them in resolving the issue, but it is just a step.